src/taco-utils/utilHelper.ts

```
@@ -320,6 +320,36 @@ module TacoUtility {
         }
 
         /**
+         * parses a JSON file which could get modified by user or another tool
+         * handles different encoding
+         * @param {string} path to the json file
+         * @return {any} parsed JSON object
+         */
+        public static parseUserJSON(filePath: string): any {
+            var contents: Buffer = fs.readFileSync(filePath);
+            // try the simplest path first
+            try {
+                return JSON.parse(contents.toString());
+            } catch (ex) {
+                UtilHelper.emptyMethod();
+            }
+
+            // may be this is a UTF-8 with BOM
+            var iconv = require("iconv-lite");
+            try {
+                return JSON.parse(iconv.decode(contents, "utf8"));
+            } catch (ex) {
+                UtilHelper.emptyMethod();
+            }
+            // May be this is a UTF-16 file
+            try {
+                return JSON.parse(iconv.decode(contents, "utf16"));
+            } catch (ex) {
+                UtilHelper.emptyMethod();
+            }
```

I think we should throw an exception if we can't parse the file...